### ISSUE TITLE
Expose additional model metadata settings

### ISSUE THREAD

**Issue by hev** [hev/vibecheck#16]
## Overview

Add support for all OpenRouter API parameters as optional metadata settings in evaluation suite YAML files. This will allow users to fine-tune model behavior directly in their eval configurations.

## Reference

Based on the [OpenRouter API Parameters documentation](https://openrouter.ai/docs/api-reference/parameters), the following parameters should be added as optional metadata settings.

### Sampling Parameters

- **`temperature`** (float, 0.0-2.0, default: 1.0)
  - Controls response variety. Lower = more predictable, higher = more diverse.

- **`top_p`** (float, 0.0-1.0, default: 1.0)
  - Limits token choices to top tokens whose probabilities sum to P. Lower = more predictable.

- **`top_k`** (integer, ≥0, default: 0)
  - Limits model's token choices at each step. Value of 1 = always picks most likely token.

- **`min_p`** (float, 0.0-1.0, default: 0.0)
  - Minimum probability for a token relative to the most likely token.

- **`top_a`** (float, 0.0-1.0, default: 0.0)
  - Consider only tokens with "sufficiently high" probabilities based on the most likely token.

### Penalty Parameters

- **`frequency_penalty`** (float, -2.0 to 2.0, default: 0.0)
  - Reduces repetition based on token frequency in input. Scales with occurrences.

- **`presence_penalty`** (float, -2.0 to 2.0, default: 0.0)
  - Reduces repetition of specific tokens already used. Does not scale with occurrences.

- **`repetition_penalty`** (float, 0.0-2.0, default: 1.0)
  - Reduces repetition of tokens from input. Scales based on original token probability.

### Generation Control

- **`max_tokens`** (integer, ≥1)
  - Upper limit for number of tokens the model can generate.

- **`seed`** (integer)
  - Enables deterministic sampling. Same seed + parameters = same result (not guaranteed for all models).

- **`stop`** (array of strings)
  - Stop generation immediately if model encounters any token in the stop array.

### Output Format

- **`response_format`** (object)
  - Forces specific output format. `{ "type": "json_object" }` enables JSON mode.

- **`structured_outputs`** (boolean)
  - If model can return structured outputs using response_format json_schema.

### Logging & Debugging

- **`logprobs`** (boolean)
  - Whether to return log probabilities of output tokens.

- **`top_logprobs`** (integer, 0-20)
  - Number of most likely tokens to return at each position with log probabilities. Requires `logprobs: true`.

### Tool Calling

- **`tools`** (array)
  - Tool calling parameter, following OpenAI's tool calling request shape.

- **`tool_choice`** (string or object)
  - Controls which tool is called: `'none'`, `'auto'`, `'required'`, or specific tool object.

- **`parallel_tool_calls`** (boolean, default: true)
  - Whether to enable parallel function calling during tool use.

### Advanced

- **`logit_bias`** (map/object)
  - JSON object mapping token IDs to bias values (-100 to 100). Added to logits before sampling.

- **`verbosity`** (enum: "low", "medium", "high", default: "medium")
  - Controls verbosity and length of model response.

### Schema Updates

Update `EvalSuiteMetadataSchema` in `packages/cli/src/types.ts` to include all these parameters as optional fields.

### YAML Example

```yaml
metadata:
  name: my-eval-suite
  model: anthropic/claude-3.5-sonnet
  system_prompt: "You are a helpful assistant"
  temperature: 0.7
  top_p: 0.9
  max_tokens: 500
  seed: 42
  stop: ["\n\n", "Human:"]
  response_format:
    type: json_object
  tools: [...]
  tool_choice: auto
```

### CLI Override Support

Consider adding CLI flags for commonly used parameters (e.g., `--temperature`, `--max-tokens`) to allow runtime overrides similar to existing `--model` and `--system-prompt` flags.

### Validation

- Ensure proper type validation for each parameter
- Validate ranges where applicable (e.g., temperature 0.0-2.0)
- Handle provider-specific parameter support gracefully

## Benefits

- Fine-grained control over model behavior in eval configurations
- Support for deterministic testing via `seed`
- Better control over output format (JSON mode, structured outputs)
- Enhanced tool calling configuration
- More comprehensive model parameter testing

## Related

- Current metadata schema: `packages/cli/src/types.ts`
- OpenRouter API Reference: https://openrouter.ai/docs/api-reference/parameters
